### PR TITLE
added github workflow for unit tests

### PR DESCRIPTION
**Pull request by willarmiros**
## Which problem is this PR solving?

- https://github.com/open-telemetry/opentelemetry-js/issues/1259

## Short description of the changes

Mostly just a copy of #272, with added paths for caching. I'm leaving CircleCI config in for now to allow both tests to run in parallel, but will remove them once I confirm everything passes on GH Actions. From there I believe it's up to the maintainers to remove CircleCI checks from branch protection & disable the workflow on CircleCI.

Next steps are to migrate container images test images from CircleCI to GHCR, then make a CD workflow using GH Actions which will require more discussion.

### CODE REVIEW

**Comment by willarmiros** [comment]
Fixed coverage to only run once (on Node 12) and removed CircleCI (which caused the CI failure). This should be ready for review!

**Review by johnbley — APPROVED** [review]
Looks like a good port to me, nice work.

**Comment by willarmiros** [comment]
Thanks @johnbley! I have no idea why the codecov is dropping on this, I thought running coverage on all Node versions instead of just Node 12 might help, but I guess not. Any ideas/does this impact merging?

**Review by vmarchaud — APPROVED** [review]

**Comment by johnbley** [comment]
> Thanks @johnbley! I have no idea why the codecov is dropping on this, I thought running coverage on all Node versions instead of just Node 12 might help, but I guess not. Any ideas/does this impact merging?

It doesn't look like a real problem to me, possibly just a blip owing to something not lining up right.  The root checkout directories changed from`/root/project/detectors/node/` to  `/__w/` and that might be confusing the codecov report?  In any case it doesn't show any specific instances of actual decreased coverage, so I'm not particularly worried.

File: .github/workflows/unit-test.yml [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+            propagators/*/node_modules
+          key: ${{ runner.os }}-${{ matrix.container }}-${{ hashFiles('**/package.json') }}
+      - name: Install Root Dependencies
+        run: npm install --ignore-scripts
+      - name: Boostrap Dependencies
+        run: npx lerna bootstrap --no-ci
+      - name: Unit tests
+        run: npm run test
+      - name: Report Coverage
+        run: npm run codecov
```
Comment (dyladan) [inline .github/workflows/unit-test.yml:99]: Looks like this will report coverage for each version of node, not just 14

Comment (willarmiros) [reply .github/workflows/unit-test.yml:99]: Oh yeah I removed the conditional here to see if it would fix the codecov diff, I'll add it back in.

Comment (dyladan) [reply .github/workflows/unit-test.yml:99]: I'm going to approve with the assumption you will fix this before it gets merged

**Review by dyladan — APPROVED** [review]

File: .github/workflows/unit-test.yml (hunk 1 above)
Comment (willarmiros) [reply .github/workflows/unit-test.yml:100]: Should be good now :) thanks for the catch

**Pull request merged**